### PR TITLE
JG/Authentication fixed

### PR DESCRIPTION
**Pull request by joegor85**
Fixed problem with logging in not working and unhashed passwords being stored in DB. Now passwords will be encrypted, hashed, and salted correctly in the database and you can correctly log in.

Also, created a request.http file and was modifying it just for testing.

The testEnv.js file was also modified and I thought we had changed it to localhost:3000 to get it to work. I don't think that will affect anything, but if it does, feel free to kick it back and I can change it back to 8000.

### CODE REVIEW

File: api/src/request.http
Comment (ranegray) [inline api/src/request.http:1]: Do we need this file going forward? If not, you can delete it.

File: api/src/routesUsers.js
Comment (ranegray) [inline api/src/routesUsers.js:1]: Can we remove all the console.log's from this file if we've got it working?

File: api/src/routesUsers.js [hunk 1]
```diff
@@ -72,7 +74,7 @@ async function create(req, res, next) {
       const result = await db
         .query(
           `INSERT INTO users (${keys}) VALUES ($1, $2, $3, $4, $5) RETURNING *`,
-          [first_name, last_name, email, salt, hash]
+          [first_name, last_name, email, Buffer.from(salt, 'hex'), hash]
```
Comment (ranegray) [inline api/src/routesUsers.js:77]: What does the Buffer do?

File: api/src/routesUsers.js [hunk 2]
```diff
@@ -154,10 +156,12 @@ async function authenticate(req, res, next) {
   }
 
   const storedHash = result.rows[0].password_hash;
-  const storedSalt = result.rows[0].salt;
-
-  const inputHash = crypto.pbkdf2Sync(password, storedSalt, 10000, 64, 'sha512').toString('hex');
+  // console.log(storedHash);  //for testing/troubleshooting
+  const storedSalt = result.rows[0].salt.toString('hex');
```
Comment (ranegray) [inline api/src/routesUsers.js:160]: Are we still storing the salt? 

If so, we shouldn't need to do that.

**Review by ranegray — COMMENTED** [review]
Let's clean it up a little bit. Maybe add some comments to help us understand what's going on in different functions. The code isn't really self explanatory. 

Also do we plan on refactoring to a more robust backend? Using bcrypt and better utilized jwts.

**Comment by ranegray** [comment]
Let's reference this repo for our authentication setup, [Node API design](https://github.com/Hendrixer/api-design-v4-course/tree/main)

File: api/src/routesUsers.js (hunk 1 above)
Comment (joegor85) [reply api/src/routesUsers.js:77]: The buffer creates a buffer object that converts the salt to binary from hexadecimal representation. After all my tinkering, this was the only way I could get it to function properly.

**Pull request merged**